### PR TITLE
feat: Add URL moderation allowlist

### PR DESCRIPTION
**Pull request by zunderscore**
### Description of the Change
Adds an allowlist feature to URL moderation to automatically permit links that match allowlist rules. Also updated URL regex to match URLs that don't have trailing `/` after domain name (e.g. `https://firebot.app` vs `https://firebot.app/`).

### Applicable Issues
#1564

### Testing
- Added terms to the URL allowlist one at a time
- Imported terms en masse to the URL allowlist from a text file
- Deleted individual terms from the URL allowlist 
- Deleted entire URL allowlist at once
- Verified URL allowlist DB file saves
- Verified URL allowlist reloads when reopening Firebot
- Tested moderation with alternate account
  - Message with single URL that matched a rule allowed
  - Message with single URL that did NOT match a rule deleted
  - Message with multiple URLs that all matched rules allowed
  - Message with multiple URLs where at least one URL did NOT match a rule deleted

### Screenshots
image
image

### CODE REVIEW

File: gui/app/directives/modals/misc/editUrlAllowlistModal.js [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+                <h4 class="modal-title">Edit URL Allowlist</h4>
+            </div>
+            <div class="modal-body">
+                <p class="muted" style="margin-bottom:10px;">URLs found in messages containing any words or phrases listed here will be automatically allowed.</p>
+                <p class="muted" style="margin-bottom:20px;">NOTE: If multiple URLs are found in a message and ANY of them are not allowed, the entire message will be deleted.</p>
+                <div style="margin: 0 0 25px;display: flex;flex-direction: row;">
+
+                    <div class="dropdown">
+                        <button class="btn btn-primary dropdown-toggle" type="button" id="add-options" data-toggle="dropdown" aria-haspopup="true" aria-expanded="true">
+                            <span class="dropdown-text"><i class="fas fa-plus-circle"></i> Add Word(s)</span>
```
Comment (ebiggz) [inline gui/app/directives/modals/misc/editUrlAllowlistModal.js:21]: In this context I think this should be updated to `Add URL(s)`

File: gui/app/directives/modals/misc/editUrlAllowlistModal.js [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+                            <span class="caret"></span>
+                        </button>
+                        <ul class="dropdown-menu" aria-labelledby="add-options">
+                            <li role="menuitem" ng-click="$ctrl.addWord()"><a href style="padding-left: 10px;"><i class="fad fa-plus-circle" style="margin-right: 5px;"></i> Add single word</a></li>
+                            <li role="menuitem" ng-click="$ctrl.showImportModal()"><a href style="padding-left: 10px;"><i class="fad fa-file-import" style="margin-right: 5px;"></i> Import from .txt file <tooltip text="'Import a list of words/phrases from a txt file'"></tooltip></a></li>
```
Comment (ebiggz) [inline gui/app/directives/modals/misc/editUrlAllowlistModal.js:26]: Update "words/phrases" to URLs

File: gui/app/directives/modals/misc/editUrlAllowlistModal.js [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+                        </ul>
+                    </div>
+
+                    <div style="display: flex;flex-direction: row;justify-content: space-between;margin-left: auto;">
+                        <searchbar placeholder-text="Search words..." query="$ctrl.search" style="flex-basis: 250px;"></searchbar>
```
Comment (ebiggz) [inline gui/app/directives/modals/misc/editUrlAllowlistModal.js:31]: "Search URLs..."

**Review by ebiggz — CHANGES_REQUESTED** [review]
Couple places where we need to update references to "words" to "URLs" but otherwise looks good!

File: gui/app/directives/modals/misc/editUrlAllowlistModal.js (hunk 1 above)
Comment (zunderscore) [reply gui/app/directives/modals/misc/editUrlAllowlistModal.js]: Fixed

File: gui/app/directives/modals/misc/editUrlAllowlistModal.js (hunk 2 above)
Comment (zunderscore) [reply gui/app/directives/modals/misc/editUrlAllowlistModal.js]: Fixed

File: gui/app/directives/modals/misc/editUrlAllowlistModal.js (hunk 3 above)
Comment (zunderscore) [reply gui/app/directives/modals/misc/editUrlAllowlistModal.js]: Fixed

**Review by SReject — APPROVED** [review]

**Review by ebiggz — APPROVED** [review]

**Pull request merged**